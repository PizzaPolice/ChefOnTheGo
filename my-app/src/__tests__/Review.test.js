import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, mount, render } from 'enzyme';
import Review from '../components/review.jsx'

it('Review component renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Review />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

// This test is shallow rendering our Review component
describe('Review Component', () => {
    it('Review should render without throwing an error', () => {
        expect(shallow(<Review />).exists(<form className='Review'></form>)).toBe(true)
      })
      it("should render initial layout", () => {
        const component = shallow(<Review />);
        expect(component.getElements()).toMatchSnapshot();
        })
    });
