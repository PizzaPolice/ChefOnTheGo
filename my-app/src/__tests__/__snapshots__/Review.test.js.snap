// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Review Component should render initial layout 1`] = `
Array [
  <div>
    <h1>
        Thank you for using Chef on the Go!
    </h1>
    <h2>
        Please let us know about your experience Below
    </h2>
    <div
        className="rating"
    >
        <form>
            <h4>
                My Experience was ...
            </h4>
            <label>
                Absolutely Terrible
                <input
                    name="star"
                    type="radio"
                    value="1"
                />
            </label>
            <label>
                Sub-par
                <input
                    name="star"
                    type="radio"
                    value="2"
                />
            </label>
            <label>
                Good
                <input
                    name="star"
                    type="radio"
                    value="3"
                />
            </label>
            <label>
                Very Good
                <input
                    name="star"
                    type="radio"
                    value="4"
                />
            </label>
            <label>
                Amazing
                <input
                    name="star"
                    type="radio"
                    value="5"
                />
            </label>
        </form>
        <textarea
            cols="70"
            defaultValue="Enter your review here! Be sure to be descriptive, It was good tells fellow connoisseurs almost nothing about the chef"
            name="How was your experience?"
            rows="10"
        />
        <div>
            <form
                onSubmit={[Function]}
            >
                <input
                    className="review"
                    type="submit"
                    value="Submit"
                />
                <input
                    className="noreview"
                    type="submit"
                    value="I don't want to leave a review"
                />
            </form>
        </div>
    </div>
</div>,
]
`;
